lib/taxonomy/associations_translation.rb

```diff
@@ -0,0 +1,42 @@
+module Taxonomy::AssociationsTranslation
+  def self.mapped_taxon_content_ids_for_edition(edition)
+    [
+      # This method still uses the legacy "specialist sectors" name
+      edition.specialist_sectors,
+      # topics means policy areas in this context
+      (edition.topics if edition.can_be_associated_with_topics?),
+      (edition.policies if edition.can_be_related_to_policies?)
+    ].compact.flatten.flat_map do |legacy_taxon|
+      content_ids = fetch_topic_taxonomy_taxons_content_ids(
+        legacy_taxon.content_id
+      )
+
+      log_mapping(legacy_taxon, content_ids)
```

Comment:
At least to me, it's quite important to be able to know what is going on, and be able to lookup specific cases where this code has done something, so I'm for keeping it. I realise that logging is a pretty bad way of storing data in general, but it seems ok to me for a feature like this.

Reply:
OK. Can we tweak the logging slightly so it can be separated from the main pipeline? Logging is helpful but it's making the logic harder for me to follow.

Reply:
I'm not quite sure what I can do with the logging code here. The information it logs is available at this specific point, and not any other as far as I can see, which makes moving it difficult

Reply:
I'm happy it's more readable now - thanks for extracting that method.